docs/DeveloperGuide.md

```diff
@@ -1,29 +1,115 @@
+[DeveloperGuide.md](DeveloperGuide.md)
 # Developer Guide
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
 
```

Comment:
Should be better to mention the main components of the app architecture to understand the overview of components used?